src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1629,7 +1629,16 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener)
           }
           PullRequestSCMRevision prRev =
               createPullRequestSCMRevision(pr, prhead, listener, ghRepository);
-          prRev.validateMergeHash();
+          try {
```

Comment:
We only need to do this if the `prHead`  is a merge, right? 
Maybe encapsulate this in a method and call from the two different places we currently call `validateMergeHash()`?
`getValidatedMergeOrNull()`?